interfaces/apparmor/spec.go

```diff
@@ -399,10 +433,45 @@ func (spec *Specification) SecurityTags() []string {
 			tags = append(tags, t)
 		}
 	}
+	for t := range spec.parametricSnippets {
+		if !seen[t] {
+			tags = append(tags, t)
+		}
+	}
 	sort.Strings(tags)
 	return tags
 }
 
+func (spec *Specification) snippetsForTag(tag string) []string {
```

Comment:
Unit testing this in isolation would be nice, but the iio tests do cover nearly all of it, so not a blocker from me (though, 'case 0' and 'len(spec.parametricSnippets[tag]) == 0' are not AFAICS).

Reply:
I've added lots of extra unit tests now.